From 4d0ed44419d97502eb0ed6210cc2af7e5e553e60 Mon Sep 17 00:00:00 2001
From: Tim Blechmann <tim.blechmann@qt.io>
Date: Fri, 12 Sep 2025 11:32:57 +0800
Subject: [PATCH 083/553] CMake: remove no_core_dep relic

Bundled zlib is linked transitively from QtCore when usid from any
consumer library. However there is an "AND NOT no_core_dep" condition,
though no_core_dep does not seem to be defined anywhere and seems to be
legacy.
Removing to make the code easier reason about.

Change-Id: If413a9c9ff9a816e49a32d70f9d1bae16e11ed25
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
(cherry picked from commit 0e4ea3969caadf23dff11453d9a0f911a028f420)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/3rdparty/freetype/CMakeLists.txt | 2 +-
 src/3rdparty/libpng/CMakeLists.txt   | 2 +-
 src/gui/CMakeLists.txt               | 2 +-
 src/network/CMakeLists.txt           | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/3rdparty/freetype/CMakeLists.txt b/src/3rdparty/freetype/CMakeLists.txt
index 720a98bee15..2749a6e40be 100644
--- a/src/3rdparty/freetype/CMakeLists.txt
+++ b/src/3rdparty/freetype/CMakeLists.txt
@@ -109,7 +109,7 @@ qt_internal_extend_target(BundledFreetype CONDITION NOT QT_FEATURE_system_zlib
         ../zlib/src
 )
 
-qt_internal_extend_target(BundledFreetype CONDITION NOT QT_FEATURE_system_zlib AND NOT no_core_dep
+qt_internal_extend_target(BundledFreetype CONDITION NOT QT_FEATURE_system_zlib
     LIBRARIES
         Qt::Core
 )
diff --git a/src/3rdparty/libpng/CMakeLists.txt b/src/3rdparty/libpng/CMakeLists.txt
index 49b2af24336..8b9a0430df8 100644
--- a/src/3rdparty/libpng/CMakeLists.txt
+++ b/src/3rdparty/libpng/CMakeLists.txt
@@ -54,7 +54,7 @@ qt_internal_extend_target(BundledLibpng CONDITION NOT QT_FEATURE_system_zlib
         ../zlib/src
 )
 
-qt_internal_extend_target(BundledLibpng CONDITION NOT QT_FEATURE_system_zlib AND NOT no_core_dep
+qt_internal_extend_target(BundledLibpng CONDITION NOT QT_FEATURE_system_zlib
     LIBRARIES
         Qt::Core
 )
diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index d864b4ea3d6..1ce10a3e850 100644
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -882,7 +882,7 @@ qt_internal_extend_target(Gui CONDITION NOT QT_FEATURE_system_zlib
         ../3rdparty/zlib
 )
 
-qt_internal_extend_target(Gui CONDITION NOT QT_FEATURE_system_zlib AND NOT no_core_dep
+qt_internal_extend_target(Gui CONDITION NOT QT_FEATURE_system_zlib
     LIBRARIES
         Qt::Core
 )
diff --git a/src/network/CMakeLists.txt b/src/network/CMakeLists.txt
index dae0162b90b..900b600df8b 100644
--- a/src/network/CMakeLists.txt
+++ b/src/network/CMakeLists.txt
@@ -170,7 +170,7 @@ qt_internal_extend_target(Network CONDITION NOT QT_FEATURE_system_zlib
         ../3rdparty/zlib
 )
 
-qt_internal_extend_target(Network CONDITION NOT QT_FEATURE_system_zlib AND NOT no_core_dep
+qt_internal_extend_target(Network CONDITION NOT QT_FEATURE_system_zlib
     LIBRARIES
         Qt::Core
 )
-- 
2.51.2

